runtime/databricks/automl_runtime/forecast/prophet/model.py

```diff
@@ -225,15 +253,21 @@ def model_predict(self, df: pd.DataFrame, horizon: int = None) -> pd.DataFrame:
         result_df["ts_id"] = str(df["ts_id"].iloc[0])
         return result_df[return_cols]
 
-    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, input_df: pd.DataFrame) -> pd.DataFrame:
+    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, model_input: pd.DataFrame) -> pd.DataFrame:
         """
         Predict API from mlflow.pyfunc.PythonModel
         :param context: A :class:`~PythonModelContext` instance containing artifacts that the model
                         can use to perform inference.
-        :param input_df: Input dataframe
+        :param model_input: Input dataframe
         :return: A pd.DataFrame with the forecast components.
         """
-        return self._predict_impl(input_df)
+        self._validate_cols(model_input, self._id_cols + [self._time_col])
+        model_input["ts_id"] = model_input[self._id_cols].agg('-'.join, axis=1)
```

Comment:
ditto, can we create a separate copy `X` instead of directly modifying the `model_input`?